collector/container/rhtap-slim.Dockerfile

```diff
@@ -0,0 +1,151 @@
+ARG BUILD_DIR=/build
+ARG SRC_ROOT_DIR=${BUILD_DIR}
+ARG CMAKE_BUILD_DIR=${SRC_ROOT_DIR}/cmake-build
+ARG USE_VALGRIND=false
+ARG ADDRESS_SANITIZER=false
+
+# TODO(ROX-20312): we can't pin image tag or digest because currently there's no mechanism to auto-update that.
+# TODO(ROX-20651): Use RHEL/ubi base image when entitlement is solved.
+# RPMs requiring entitlement: bpftool, cmake-3.18.2-9.el8, elfutils-libelf-devel, tbb-devel, c-ares-devel, jq-devel
+# FROM registry.access.redhat.com/ubi8/ubi:latest AS builder
+FROM quay.io/centos/centos:stream9 AS builder
+
+COPY . .
+
+# TODO(ROX-20234): use hermetic builds when installing/updating RPMs becomes hermetic.
+RUN dnf -y update \
+    && dnf -y install --nobest \
+        autoconf \
+        automake \
+        binutils-devel \
+        bison \
+        ca-certificates \
+        clang-15.0.7 \
+        cmake \
+        cracklib-dicts \
+        diffutils \
+        elfutils-libelf-devel \
+        file \
+        flex \
+        gcc \
+        gcc-c++ \
+        gdb \
+        gettext \
+        git \
+        glibc-devel \
+        libasan \
+        libubsan \
+        libcap-ng-devel \
+        libcurl-devel \
+        libtool \
+        libuuid-devel \
+        make \
+        openssh-server \
+        openssl-devel \
+        patchutils \
+        passwd \
+        pkgconfig \
+        rsync \
+        tar \
+        unzip \
+        valgrind \
+        wget \
+        which \
+        bpftool \
+    && dnf clean all
+
+ARG BUILD_DIR
+ENV BUILD_DIR=${BUILD_DIR}
+
+ARG SRC_ROOT_DIR
+ENV SRC_ROOT_DIR=${SRC_ROOT_DIR}
+
+ARG CMAKE_BUILD_DIR
+ENV CMAKE_BUILD_DIR=${CMAKE_BUILD_DIR}
+
+# TODO(ROX-20240): CMAKE_BUILD_TYPE should probably not be Release for PR, normal branch builds
+ARG CMAKE_BUILD_TYPE=Release
+ENV CMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+
+# Appends an argument to the driver download URL that is used for filtering alerts on missing kernels.
+# TODO(ROX-20240): This needs to be true on PRs only.
+ARG COLLECTOR_APPEND_CID=false
+ENV COLLECTOR_APPEND_CID=${COLLECTOR_APPEND_CID}
+
+WORKDIR ${BUILD_DIR}
+
+RUN mkdir kernel-modules \
+    && cp -a /builder builder \
+    && cp -a /collector collector \
+    && cp -a /falcosecurity-libs falcosecurity-libs \
+    && cp -a /builder/third_party third_party \
+    && cp -a /kernel-modules/MODULE_VERSION kernel-modules/MODULE_VERSION \
+    && cp -a /CMakeLists.txt CMakeLists.txt
+
+# WITH_RHEL_RPMS controls for dependency installation, ie if they were already installed as RPMs.
+# Setting the value to empty will cause dependencies to be downloaded from repositories or accessed in submodules and compiled.
+# TODO(ROX-20651): Set ENV WITH_RHEL_RPMS=true when RHEL RPMs can be installed to enable hermetic builds.
+RUN ./builder/install/install-dependencies.sh \
+    && cmake -DDISABLE_PROFILING=ON -S "${SRC_ROOT_DIR}" -B "${CMAKE_BUILD_DIR}" \
+    && cmake --build "${CMAKE_BUILD_DIR}" --target all -- -j "$(nproc)" \
```

Comment:
The problem is that `nproc` command is not aware of cgroups and hence does not know how many CPUs were actually allocated to the container. It just reports the number of CPUs seen by the kernel, this will be the number of cores of a node which are shared between all containers.

As we saw in the discussions related to RHTAP performance, CPU limits are still there and node CPUs are (currently) getting much oversubscribed. It may not cause a huge context switching overhead to let the command work with `nproc` result which is `8`, but we're certainly not getting what we asked for.

I suggest to replace this misleading `nproc` with just a fixed value (e.g. via a named variable) which we could manually tune if/when needed.

`build-collector.sh` used to set it to `2`, by the way.
https://github.com/stackrox/collector/blob/acf241f5fd43b5e9c995bbb253a866347d064bc1/builder/build/build-collector.sh#L31

Reply:
Adapted to what was configured for downstream in https://gitlab.cee.redhat.com/stackrox/rhacs-midstream/-/merge_requests/2278, implementing the suggestion of a named variable with a default.